### PR TITLE
build: enable MSVC level 3 warnings

### PR DESCRIPTION
**Pull request by dundargoc**
MSVC has 4 different warning levels: 1 (severe), 2 (significant), 3
(production quality) and 4 (informational). Enabling level 3 warnings
mostly revealed conversion problems, similar to GCC/clang -Wconversion
flag.

### CODE REVIEW

**Comment by clason** [comment]
Can you add some comment about the effect of and rationale for this change?

**Comment by dundargoc** [comment]
Good point. Thanks for the reminder.

File: src/nvim/eval/vars.c [hunk 1]
```diff
@@ -696,7 +696,7 @@ static char *ex_let_one(char *arg, typval_T *const tv, const bool copy, const bo
 
       if (!failed) {
         if (opt_type != gov_string || s != NULL) {
-          char *err = set_option_value(arg, n, s, scope);
+          char *err = set_option_value(arg, (const long)n, s, scope);
```
Comment (justinmk) [inline src/nvim/eval/vars.c:699]: seems weird that "const" needed in a value cast

Comment (dundargoc) [reply src/nvim/eval/vars.c:699]: You're right, it's not needed. Thanks.

**Pull request merged**